plugin/src/location.ts

```diff
@@ -0,0 +1,43 @@
+import GoogleAnalyticstModal from "@web/extensions/location/modals/googleAnalyticsModal";
+import TerrainModal from "@web/extensions/location/modals/terrainModal";
+import { PostMessageProps, MouseEventData } from "@web/extensions/location/types";
+
+import html from "../dist/web/location/core/index.html?raw";
+
+const reearth = (globalThis as any).reearth;
+
+reearth.ui.show(html);
+
+reearth.on("mousemove", (mousedata: MouseEventData) => {
+  reearth.ui.postMessage(
+    {
+      type: "mousedata",
+      payload: mousedata,
+    },
+    "*",
+  );
+});
+reearth.on("cameramove", () => {
+  reearth.ui.postMessage({
+    type: "getLocations",
+    payload: {
+      point1: reearth.scene.getLocationFromScreenPosition(
+        reearth.viewport.width / 2,
+        reearth.viewport.height - 1,
+      ),
+      point2: reearth.scene.getLocationFromScreenPosition(
+        reearth.viewport.width / 2 + 1,
+        reearth.viewport.height - 1,
+      ),
+    },
+  });
+});
+reearth.on("message", ({ action }: PostMessageProps) => {
+  if (action === "modal-google-open") {
+    reearth.modal.show(GoogleAnalyticstModal);
+  } else if (action === "modal-Terrain-open") {
```

Comment:
Make sure you keep to your own standard. First you have `modal-google-open` (all lowercase), and then you do `modal-Terrain-open`, followed by `modal-close`. It is best to keep everything lowercase

Reply:
Done - 557f196a036be1dd615517dd0ae44ab6ee43911e

Reply:
for eventListener - Done - 756322989fa9a1d1facaf76a678d211c363fead7